code/src/pages/MoviePage.js

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect, useState } from "react";
+
+import { Link, useParams } from "react-router-dom"; 
+import { apiKey } from "../key";
+import BackArrow from "../assets/back-arrow.svg";
+import { NotFound } from "./NotFound";
+
+export const MoviePage = () => { 
+  const { movieId } = useParams (); 
+  const [movie, setMovie] = useState ([]);
+
+  useEffect (() => { 
+    const moviePageUrl=`https://api.themoviedb.org/3/movie/${movieId}?api_key=${apiKey}&language=en-US`
+    fetch (moviePageUrl)
+      .then((response) => response.json())
+      .then((json) => {
+        setMovie(json)
+      })
+      .catch((error) => {
+        console.error('Request failed', error)
+      })
+  }, [movieId]);
+
+  if (!movie.id) {
+    return (
+      <NotFound/>
+    );
+  };
+
+return (
+  <>
```

Comment:
ooh! i looked this up. thanks for teaching me something!

Reply:
Ah nice! 🤩